robottelo/api/utils.py

```diff
@@ -550,3 +550,27 @@ def configure_provisioning(org=None, loc=None):
         'host_group': host_group.name,
         'domain': domain.name,
     }
+
+
+def wait_for_tasks(query, search_rate=1, max_tries=10):
+    """Search for tasks by specified search query and poll them to ensure that
+    task has finished.
+
+    :param query: Search query that will be passed to API call.
+    :param search_rate: Delay between searches.
+    :param max_tries: How many times search should be executed.
+    :raises: ``AssertionError``. If not tasks were found until timeout.
+    :return: List of ``nailgun.entities.ForemanTasks`` entities.
+    """
+    for _ in range(max_tries):
+        tasks = entities.ForemanTask().search(query={'search': query})
```

Comment:
Think to exclude any confusion, if you are using query this string must be:
1. 
```python
tasks = entities.ForemanTask().search(query=query)
```
2. But we you want to use this only for search, the function must be
```python 
def wait_for_tasks(search, search_rate=1, max_tries=10):
....
     tasks = entities.ForemanTask().search(query={'search': search})
....
```
but I prefer the first one, as it gives more functionality